### DISCUSSION TITLE
How does excerpt's extract option work?

### DISCUSSION THREAD

**Discussion by ouuan** [ElMassimo/iles#162] [Q&A]
There seems to be no documentation for the `extract` option of `@islands/excerpt`. I tried reading the source code, but it seems that the default behavior uses `ast` but the `extract` option only accepts `vfile` so it seems that I cannot even use this option to reproduce the default behavior.

https://github.com/ElMassimo/iles/blob/499739658b3248e2fa046b2d46cf93c5699c32d4/packages/excerpt/src/rehype-plugin.ts#L13-L18

I'm interested in this option because I was trying to remove duplicated math in the excerpt. The excerpt of "$f_i$" is now `fif_ifi​` when using KaTeX SSR that produces both MathML and CHTML outputs. It will be easy to remove the duplicated math if I have access to the HTML nodes.

**Comment by ElMassimo** [comment]
@ouuan I don't see any downsides in passing `ast` to `extract`, as a third parameter to ensure backwards compatibility.

PRs are welcome!

**Comment by ouuan** [comment]
Does the `extract` option affect Markdown components with the `excerpt` prop? Though I don't need to change the behavior of the Markdown component with the `excerpt` prop, I thought that the two types of excerpts should be consistent, but then I found that the `extract` option returns plain text and felt confused. Maybe this behavior is OK but there should be some documentation for it.

**Reply by ElMassimo** [reply]
No, `extract` is meant to get a text-only excerpt that can be used anywhere (page description, RSS feed), while the component version can render components and anything you add prior to the "break".

The "breaking point" for the component `excerpt` is found using [`isSeparator`](https://github.com/ElMassimo/iles/blob/499739658b3248e2fa046b2d46cf93c5699c32d4/packages/excerpt/src/rehype-plugin.ts#L19), which you can [configure using `separator`](https://github.com/ElMassimo/iles/blob/499739658b3248e2fa046b2d46cf93c5699c32d4/packages/excerpt/src/types.ts#L4-L22).

You can see the [types for the module options](https://github.com/ElMassimo/iles/blob/499739658b3248e2fa046b2d46cf93c5699c32d4/packages/excerpt/src/types.ts#L4-L22) as a reference, but I agree that documentation is needed around these advanced use cases.

At this point I think each îles module warrants its own page in the docs, if you have any suggestions for an _Excerpts_ page, PRs are welcome 😃
